### ISSUE TITLE
Exclude folders and file types

### ISSUE THREAD

**Issue by mpariente** [jellyedwards/gitsweep#9]
Thanks for gitsweep, it's very useful !

Would it be possible to exclude files and folders based on expressions as well? Like 
```bash
*.wav
*.[oa]  #like the example is .git/info/exclude
folder/path/*
...
```

It would be amazing !

**Comment by jellyedwards** [comment]
You're welcome, glad you like it!!

I'm not against the idea but how would you see the interface managing it?

The purpose at the moment is to replace having to type stuff into `exclude` or a `git` command with a single click but I think I'd have to allow some sort of free text entry for expressions, and in which case why not just go directly to the `exclude` file?

**Comment by mpariente** [comment]
Thanks for your answer !

> why not just go directly to the exclude file?

That would be fine, but then the matching files should be listed in the untracked files in the interface. Currently, I can add those patterns, but the list of tracked files still show the matching files, which is confusing.

**Comment by jellyedwards** [comment]
Sorry it's taking me so long to reply, don't have much spare time!

So if you have something like 

`imgs/*`

in your `exclude` file, do you want the list of untracked files in the UNDER THE RUG window have each file listed e.g.

```
imgs/1.png
imgs/2.png
imgs/3.png
...
imgs/1000.png
```

I'm hesitant to do this because it would really clog up the listing and it's not possible to do any action (unsweep) on these.

Or do you want just the `imgs/*` pattern in the list of untracked files (it also won't have the unsweep button though)?

**Comment by mpariente** [comment]
> Or do you want just the imgs/* pattern in the list of untracked files (it also won't have the unsweep button though)?

That would be better. And why couldn't we have the button? 

And would we remove all the matching files from the tracked files?

**Comment by jellyedwards** [comment]
I guess you could, all it would do would be delete the pattern from the exclude file. Let me see what I can come up with...

**Comment by jellyedwards** [comment]
So I just tried `img/*.png` in my exclude file and it shows up as this currently:
![image](https://user-images.githubusercontent.com/14051992/158640743-a95629a6-9c61-4ea8-82c0-d2c9d8124a79.png)

With the changes I'm working on it should show this instead:
![image](https://user-images.githubusercontent.com/14051992/158640481-1dbb5aba-e8ff-45f6-98b0-877d8153abfc.png)

How does that look to you?

**Comment by mpariente** [comment]
That looks great ! 

> all it would do would be delete the pattern from the exclude file.

This would be the best IMO

And regarding the files remaining in the tracked files, do you think there is a solution?

**Comment by jellyedwards** [comment]
> regarding the files remaining in the tracked files, do you think there is a solution?

I'm going to try watching the exclude file for changes, and if it spots any tell git to refresh itself. I tried it here and it seems to take a second, but works. I'll publish soon & you can try it out!

**Comment by mpariente** [comment]
Great, thanks ! 
Is the extension automatically updated?

**Comment by jellyedwards** [comment]
It should do, it's been published a while now, v0.0.9.

If you go to the Extensions tab, click the 3 dots at the top right, you can see if you have auto update extension set:

![image](https://user-images.githubusercontent.com/14051992/159664679-b05bc29c-3eda-4866-a9bc-7c8205828673.png)

**Comment by mpariente** [comment]
Thanks :pray:
